### PR TITLE
Fix: Apply more escaping to prevent XSS vulnerabilities

### PR DESCRIPTION
**Pull request by JasonTheAdams**
## Description

This is a follow up to #6438 with even more fixes in place. Namely, this targets places where the URL is omitted, which fell back on `$_SERVER['REQUEST_URI']`, which can be considered vulnerable.

This is all related to the following wp.org effort: https://make.wordpress.org/plugins/2015/04/20/fixing-add_query_arg-and-remove_query_arg-usage/

## Affects

Large portions of the admin-side such as:
- Imports
- Exports
- Donor actions
- Donor registration emails
- Form registration
- Emails
- Stripe account onboarding

## Testing Instructions

Test out the following clear examples:
- `https://example.com/wp-admin/edit.php?post_type=give_forms&page=give-tools&a"><script>alert(/XSS/)</script>`
- `https://example.com/wp-admin/edit.php?post_type=give_forms&page=give-tools&tab=import&a"><script>alert(/XSS/)</script>`

Mostly, make sure that the affected areas continue to work without issue.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@Benunc @ravinderk I got more feedback on other places that needed escaping. I also asked if we can send them the zip for further testing. Please look this over and test it out as well. Thank you!

File: includes/admin/admin-actions.php [hunk 1]
```diff
@@ -76,10 +76,12 @@ function give_redirect_to_clean_url_admin_pages() {
 	if ( $redirect ) {
 		// Redirect.
 		wp_redirect(
-			remove_query_arg(
-				[ '_wp_http_referer', '_wpnonce' ],
-				wp_unslash( $_SERVER['REQUEST_URI'] )
-			)
+            esc_url(
```
Comment (ravinderk) [inline includes/admin/admin-actions.php:79]: @JasonTheAdams WordPress suggested using esc_url_raw when the URL is part of the HTTP header.

ref: https://make.wordpress.org/plugins/2015/04/20/fixing-add_query_arg-and-remove_query_arg-usage/

> When it’s being printed to a page (for example as a link), you should use [esc_url()](https://developer.wordpress.org/reference/functions/esc_url/). When it’s being used in HTTP headers or as part of a HTTP request (for example, as part of a location redirect header or in a [wp_remote_get()](https://developer.wordpress.org/reference/functions/wp_remote_get/) call), you should use [esc_url_raw()](https://developer.wordpress.org/reference/functions/esc_url_raw/).

Comment (JasonTheAdams) [reply includes/admin/admin-actions.php]: Good catch! I didn't think about the redirect functions. Resolved in 2877aba0

File: src/Tracking/AdminActionHandler.php [hunk 2]
```diff
@@ -101,7 +101,7 @@ public function optInToUsageTracking()
             $this->settings->saveUsageTrackingOptionValue('disabled');
         }
 
-        wp_safe_redirect(remove_query_arg('give_action'));
+        wp_safe_redirect(esc_url(remove_query_arg('give_action')));
```
Comment (ravinderk) [inline src/Tracking/AdminActionHandler.php]: fixed: https://github.com/impress-org/givewp/compare/2877aba089ec...d4bf2eb54d4a

File: src/Tracking/AdminActionHandler.php [hunk 3]
```diff
@@ -77,7 +77,7 @@ public function optOutFromUsageTracking()
 
         $this->usageTrackingOnBoarding->disableNotice($timestamp);
 
-        wp_safe_redirect(remove_query_arg('give_action'));
+        wp_safe_redirect(esc_url(remove_query_arg('give_action')));
```
Comment (ravinderk) [inline src/Tracking/AdminActionHandler.php]: fixed: https://github.com/impress-org/givewp/compare/2877aba089ec...d4bf2eb54d4a

**Review by ravinderk — APPROVED** [review]

**Review by rickalday — APPROVED** [review]
I tested all admin areas without any issues:
- Imports
- Exports
- Donor actions
- Donor registration emails
- Form registration
- Emails
- Stripe account onboarding

**Comment by JasonTheAdams** [comment]
This has been confirmed by someone from wp.org to resolve the remaining concerns. We're good to go. 👍

**Pull request merged**